### DISCUSSION TITLE
Direct Monitoring guitar with playback in stereo

### DISCUSSION THREAD

**Discussion by leehbi** [geoffreybennett/alsa-scarlett-gui#125] [Q&A] [answered]
Hi,

I have Scarlett 4th Gen 4 x4 working with Fedora 40.  

I'm recording guitar through input 1 and recording with Reaper.

The attached routing/levels let me direct monitor with stereo headphones, record a mono track in Reaper and have it played back using Mono (L&R) i.e. through both sides of headphones.

![DirectMonitor_repaer_mono_record_stero_play](https://github.com/geoffreybennett/alsa-scarlett-gui/assets/8226987/41716446-a41c-4f7a-a9c8-90e5f0fe9f3f)

I assume this is correct.   I have to mute the recording track else I get an echo/delay but I guess  this is normal.

If there is a more effective way to route this I'm interested.   Note, I want to go through the Scarlett DSP

Lee

**Comment by geoffreybennett** [comment, marked as answer]
Hi @leehbi,

If you never want to hear stereo audio out of your headphones, then that mixer setup you've got there is "good enough", but I'd suggest this setup instead:

![image](https://github.com/geoffreybennett/alsa-scarlett-gui/assets/1866657/dfbe3a84-0018-49b9-b24a-3e31837288a8)

I've never used Reaper so I can't tell you how to use it, but you should set it up so that your mono track is output to both left & right channels. I would have thought that that is the default. If you try my suggested setup and you're only hearing guitar in the left channel then I would guess that Reaper is actually doing a stereo recording, not a mono recording.

Having an echo/delay is normal if you have both direct (hardware) monitoring and software monitoring enabled at the same time. When you say "have to mute the recording track" do you actually mean hitting the Mute button on the track? Then you have to unmute the track to listen to it? If so, look for an option in Reaper to disable "input monitoring" instead.

Regards,
Geoffrey.

**Comment by leehbi** [comment]
Hi Geoffrey

Thanks for the reply - I'm newish  to Reaper - the issue that was causing
mono in phones is that Reaper defaulted to hardware outputs 1/2.
Obviously, I only have my guitar plugged into 1 :-)  I think I had in the
back of my mind that 1/2 are auto linked to 5/6.

I had to add to hardware routing for 5/6 for Master Output  (my headphones)
in Reaper.   That was kind of counter intuitive for me :)

When I playback in Reaper the Levels in your client app light up 1/2/5/6
outputs.

This client you wrote is really powerful.

And thanks for the suggestion to check for the monitoring setting - there's
a tiny little speaker icon on the track that let's me turn off Record
Monitoring.   I'm coming to Reaper from Studio One !

Best

Lee

**Comment by leehbi** [comment]
I forget to mention I was using pulse audio setting in Reaper.

I couldn't get Jack to work until I installed :
https://fedora.pkgs.org/40/fedora-updates-x86_64/pipewire-jack-audio-connection-kit-1.0.7-2.fc40.i686.rpm.html

After installing this package I was able to use JACK setting in reaper.
This gave me output in my phones on 1/2 for playback as expected.   The
latency is so much lower too when recording.

This must have have been a pulse audio-pipeware issue.

Oh well I guess Linux Audio still has some rough edges.

I'm happy now

Lee

**Comment by geoffreybennett** [comment]
Good to hear that it's working! One other thing I'd say to check (if using the PulseAudio setting, or using any other apps) is the profile selected in pavucontrol under the configuration tab. The default is a surround-sound setup which can lead to weirdness like your left headphone output getting the subwoofer (LFE) signal. I don't think that's the issue here since your routing shows PCM Outputs 5/6 being ignored, but something to keep in mind for later.
